packages/media-query/src/media-query.hook.tsx

```diff
@@ -148,3 +150,82 @@ const createMediaQueries = (breakpoints: Dict) => {
     return mediaQuery
   })
 }
+
+type BreakpointValues<T> = { [breakpoint: string]: T }
+
+/**
+ * React hook for getting the value for the current breakpoint from the
+ * provided responsive values object.
+ *
+ * @example
+ *
+ * const width = useBreakpointValue({ base: '150px', md: '250px' })
+ * return <Button width={width}>I'm {width} wide!</Button>
+ */
+export function useBreakpointValue<T = any>(values: BreakpointValues<T>) {
+  const breakpoint = useBreakpoint()
+
+  // if breakpoint is undefined, we'll return undefined
+  if (!breakpoint) return
+
+  // if we have no value for the current breakpoint, return the `base`
+  // breakpoint value
+  const baseValue = values.base
+
+  return values[breakpoint] || baseValue
```

Comment:
Do we already have tooling somewhere for finding a value from a breakpoints object that defaults to `base`? I saw `assignObject` but that wasn't quite what we needed here.

Reply:
Oh no. We need to make a decision on this. I was thinking on calling it either `base` or `xs`. Then we can update the docs and code to be consistent

Reply:
Is it just trying to figure out the right now or would they work differently?

Reply:
Let's leave it at `base` for now.